src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -144,141 +175,172 @@ def extract_values(
+        if "session.status" in self._raw_groupby:
+            reverse_lookup = {v: k for k, v in self.status_to_metric_field.items()}
+            return reverse_lookup[metric_field]
+        return None
 
-class UsersField(IntegerField):
-    name = "count_unique(user)"
+    def normalize(self, value: Scalar) -> Scalar:
+        value = super().normalize(value)
+        # In the sessions API, sum() and count_unique() return integers
+        if isinstance(value, float):
+            return int(value)
+        return value
+
+
+class SessionsField(CountField):
+    name = "sum(session)"
 
-    metric_field_to_session_status = {
-        MetricField(None, SessionMetricKey.HEALTHY_USER.value): "healthy",
-        MetricField(None, SessionMetricKey.ABNORMAL_USER.value): "abnormal",
-        MetricField(None, SessionMetricKey.CRASHED_USER.value): "crashed",
-        MetricField(None, SessionMetricKey.ERRORED_USER.value): "errored",
-        MetricField(None, SessionMetricKey.ALL_USER.value): None,
+    status_to_metric_field = {
+        SessionStatus.HEALTHY: MetricField(None, SessionMetricKey.HEALTHY.value),
+        SessionStatus.ABNORMAL: MetricField(None, SessionMetricKey.ABNORMAL.value),
+        SessionStatus.CRASHED: MetricField(None, SessionMetricKey.CRASHED.value),
+        SessionStatus.ERRORED: MetricField(None, SessionMetricKey.ERRORED.value),
+        None: MetricField(None, SessionMetricKey.ALL.value),
     }
 
-    def get_session_status(
-        self, metric_field: MetricField, raw_groupby: Sequence[str]
-    ) -> Optional[str]:
-        return self.metric_field_to_session_status[metric_field]
+    def accumulate(self, old_value: Scalar, new_value: Scalar) -> Scalar:
+        # This is only needed for a single specific scenario:
+        # When we filter by more than one session.status (e.g. crashed and abnormal),
+        # but do *not* group by session.status, we want to sum up the values from the different metrics,
+        # e.g. session.crashed + session.abnormal
+        assert isinstance(old_value, int)
+        assert isinstance(new_value, int)
+        return old_value + new_value
 
-    def get_metric_fields(self, raw_groupby: Sequence[str]) -> Sequence[MetricField]:
-        if "session.status" in raw_groupby:
-            return [
-                # Always also get ALL, because this is what we sort by
-                # in the sessions implementation, with which we want to be consistent
-                MetricField(None, SessionMetricKey.ALL_USER.value),
-                # These are the fields we actually need:
-                MetricField(None, SessionMetricKey.HEALTHY_USER.value),
-                MetricField(None, SessionMetricKey.ABNORMAL_USER.value),
-                MetricField(None, SessionMetricKey.CRASHED_USER.value),
-                MetricField(None, SessionMetricKey.ERRORED_USER.value),
-            ]
-        return [MetricField(None, SessionMetricKey.ALL_USER.value)]
+
+class UsersField(CountField):
```

Comment:
This class (and `SessionsField`) are quite specific for the query. I'd set more specific names for the classes.